### ISSUE TITLE
Use Universal Components instead of Sailfish Silica

### ISSUE THREAD

**Issue by marmistrz** [geobra/harbour-shmoose#40]
This will have the following benefits: 
1. We will no longer have to maintain the barebones desktop UI, having a full-fledged QtQuick Controls version
2. No-extra-effort Nemo Mobile compatibility in the future (when UC get to support it)

https://github.com/M4rtinK/universal-components

**Comment by geobra** [comment]
I didn't know about UC. It's worth to try them out. That can save a lot of time and double work...

**Comment by marmistrz** [comment]
This would have an extra benefit, which's a big one - if shmoose could target both Sailfish and desktop Linux, we could possibly get more people to contribute to shmoose.

**Comment by geobra** [comment]
I can do the packaging for debian. One could also think about a cross compilation for win32. But keep in mind that for all the major platforms, there are already really good xmmp clients out there.

**Comment by marmistrz** [comment]
Yes, but if we want OMEMO support, the choice is sparse right now.

**Comment by d9h20f** [comment]
Gajim and Dino work quite well with OMEMO.

**Comment by marmistrz** [comment]
I packaged the UC for Arch: https://aur.archlinux.org/packages/universal-components-git/

**Comment by muppeth** [comment]
Sorry for porbably asking in the wrong place, I see very little to no activity in the project. Is it abandoned? Proper xmpp client with muc and omemo is the only thing i think that holds me from flashing SailfishOS on my FP2.

**Comment by geobra** [comment]
Currently there are just a few developers how contribute to this project. For me speaking, I do only open source development in the winter month. So yes, there is almost no activity in the last past month. A working omemo version can be found in the xep oemo branch. This can send and receive omemo messages using lurch, but is not attached to the GUI. Additionally the architecture on how it is integrated is really bad. I plan to start a clean integration in early 2019.

**Comment by poVoq** [comment]
This would also help a lot for a Ubuntu Touch port.
